### PR TITLE
[BUGFIX] Hive temporary tables creation fix

### PR DESCRIPTION
**Pull request by jaume-ferrarons**
This PR contains minor fixes in order to get Great Expectations running in Hive.

Changes proposed in this pull request:
- Do not quote Hive table names in temporary table creation. Hive temporary table creation used the default entry quoting table names, but that's not supported by Apache Hive. 
- ~~Reduce redundant code: `self.sql_engine_dialect.name.lower()` was executed several times. The result is computed once and stored in a variable.~~
- ~~Make `DataConnectorConfig` and `DatasourceConfig` serializable. Required to be able to store Hive's data connection objects. I had to copy `to_json_dict`'s implementation, as there are circular dependencies. Fixing that, I think, should be done in a different PR.~~

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

**Comment by jaume-ferrarons** [comment]
@cla-bot check

**Comment by talagluck** [comment]
Thanks so much for the contribution, @jaume-ferrarons! We will review and be in touch.

**Comment by rdodev** [comment]
@cdkini can you please take a look at this PR?

File: great_expectations/execution_engine/sqlalchemy_batch_data.py [hunk 1]
```diff
@@ -217,11 +224,10 @@ def _create_temporary_table(
                 temp_table_name = f"{temp_table_schema_name}.{temp_table_name}"
 
             stmt = f"CREATE OR REPLACE TEMPORARY TABLE {temp_table_name} AS {query}"
-        elif dialect == GESqlDialect.MYSQL:
-            # Note: We can keep the "MySQL" clause separate for clarity, even though it is the same as the
-            # generic case.
+        elif dialect == GESqlDialect.MYSQL or dialect == GESqlDialect.HIVE:
             stmt = f"CREATE TEMPORARY TABLE {temp_table_name} AS {query}"
         elif dialect == GESqlDialect.MSSQL:
+
```
Comment (alexsherstinsky) [inline great_expectations/execution_engine/sqlalchemy_batch_data.py]: Very minor: If we do not need this blank line, let us not add it.  Thank you.

Comment (jaume-ferrarons) [reply great_expectations/execution_engine/sqlalchemy_batch_data.py]: Removed 👍

**Comment by cdkini** [comment]
@jaume-ferrarons would you happen to have some more information (or a test case) around why you updated the config class to inherit from `SerializableDictDot`? Hoping to get a better understanding of that so we can prevent future issues.

**Comment by jaume-ferrarons** [comment]
@cdkini The issues I observed occurred when storing the results of an expectation generated with the profiling when running `great_expectations suite new`. I tried to reproduce the error with the current status of `develop` but I couldn't, I guess it got fixed somehow. So, I've just reverted the changes.

Though, I've found that `GESqlDialect` is instantiated in different places, so I've moved the logic of handling the byte strings to the class and added a simple test.

**Comment by jaume-ferrarons** [comment]
@cdkini The issues I observed occurred when storing the results of an expectation generated with the profiling when running `great_expectations suite new`. I tried to reproduce the error with the current status of `develop` but I couldn't, I guess it got fixed somehow. So, I've just reverted the changes.

Though, I've found that `GESqlDialect` is instantiated in different places, so I've moved the logic of handling the byte strings to the class and added a simple test.

**Comment by jaume-ferrarons** [comment]
I was pointed out that it's better to use back quotes for table names in Hive. I've introduced that change.

I was also pointed out that there's [this SQLAlchemy method](https://docs.sqlalchemy.org/en/14/core/internals.html?highlight=dialect#sqlalchemy.sql.compiler.IdentifierPreparer) to handle the quoting. It may be useful to switch to it in the future.

**Review by donaldheppner — APPROVED** [review]

**Review by rdodev — APPROVED** [review]

**Comment by cdkini** [comment]
@jaume-ferrarons thanks for the explanation! I'm approving the PR 😄 

I'll add a quick type annotation to pass out linter checks and set it on auto-merge so we can get your changes in. Thanks so much for all the help here!

**Pull request merged**